internal/provider/validators.go

```diff
@@ -148,3 +149,23 @@ func validateConsulClusterSize(v interface{}, path cty.Path) diag.Diagnostics {
 
 	return diagnostics
 }
+
+func validateVaultClusterTier(v interface{}, path cty.Path) diag.Diagnostics {
+	var diagnostics diag.Diagnostics
+
+	// TODO: Update the validation once vaultmodels got consistent with consulmodels
```

Comment:
Seems like the comment might have to be updated to vault, not consul :-)

Reply:
It refers to [vaultmodels](https://github.com/hashicorp/hcp-sdk-go/blob/f7f0dd4b49fd46758c2d2c9a151fe3a70e5e271e/clients/cloud-vault-service/preview/2020-11-25/models/hashicorp_cloud_vault20201125_tier.go#L27-L33) being in a different structure than [consulmodels](https://github.com/hashicorp/hcp-sdk-go/blob/f7f0dd4b49fd46758c2d2c9a151fe3a70e5e271e/clients/cloud-consul-service/preview/2021-02-04/models/hashicorp_cloud_consul20210204_cluster_config_tier.go#L21)

Reply:
I'm not sure this comment is needed. While Consul and Vault tiers will move to being more consistently named, they will not necessarily have the exact same tier options, so they will always each have their own Tier model in the SDK.